From 404c1b21e0fc4185f4b102bb2e390fe890652218 Mon Sep 17 00:00:00 2001
From: Guochun Huang <hero.huang@rock-chips.com>
Date: Fri, 28 Feb 2020 10:05:43 +0800
Subject: [PATCH] drm/rockchip: upgrade initcall level of protect loader clocks

To avoid the potential parent PLL clock of display being
closed when PD is shutdown, excute protect loader clocks
earlier.

Signed-off-by: Guochun Huang <hero.huang@rock-chips.com>
Change-Id: I5125590aac51b423e7e9fdc0032da9185c450be1
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
index 79fa20878871..b67ea7eca7f1 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
@@ -1040,7 +1040,7 @@ static int __init rockchip_clocks_loader_protect(void)
 
 	return 0;
 }
-fs_initcall(rockchip_clocks_loader_protect);
+arch_initcall_sync(rockchip_clocks_loader_protect);
 
 static int __init rockchip_clocks_loader_unprotect(void)
 {
-- 
2.35.3

